epoxy-processor/src/main/java/com/airbnb/epoxy/ModelBuilderInterfaceWriter.kt

```diff
@@ -46,7 +49,7 @@ internal class ModelBuilderInterfaceWriter(
                     // can generate the interface with the proper methods later
                     viewInterfacesToGenerate
                         .putOrMerge(
-                            it,
+                            ClassKey(it, modelInfo.superClassElement),
```

Comment:
Same here, it's not clear to me why we're using the super class as the originating element. The Gradle documentation (https://docs.gradle.org/5.0/userguide/java_plugin.html#sec:incremental_annotation_processing) states: 

> When a source file is recompiled, Gradle will recompile all files generated from it. When a source file is deleted, the files generated from it are deleted.

And I believe in that context "source file" refers to elements marked as originating. So here we would want to recompile generated files when the model class is recompiled. It's super may or may not change in that case (but probably not).

Reply:
See above comment 👍

Reply:
I don't think what this ClassKey is representing is quite right. To clarify what this "view interfaces" code is doing, let's  say we have ViewA and ViewB which both implement the same "Checkable" interface. The key to this map needs to be the "Checkable" interface element, and the map should store the set of  methods that both ModelA and ModelB have in common (since we know they have a shared interface, this is a simplification we use).

By adding the  model element to the map key we will now have separate entries for each view. I think this means that technically the interface generated by this (`writeFilesForViewInterfaces`) would be aggregating, since it is influenced by multiple view classes.

This  complicates matters here unfortunately, and I don't see a good way to not have this be aggregating, since the generated interface will always depend on multiple models.  For example, say we have an interface  `Header` that  has  functions  `setTitle` and  `setCaption`.  ViewA and ViewB both implement this interface, but only ViewA marks `setTitle` as a ModelProp, and only ViewB marks `setCaption` as a ModelProp. The resulting generated models  don't actually share  any methods in common from this interface,  and the generated interface they share would actually be empty. If either model was changed to mark the other interface method as  ModelProp then the generated interface would change to include it.

This  example  is contrived, but  imagine an interface has many functions and  views only want their models to contain a subset of those.  The point of the model interface is to provide a generated interface to the user that represents the  shared subset of model methods.

I don't see any good ways to avoid aggregation here, unless maybe we change it to be a processor error if multiple models with the same  view interface don't have the  same interface methods marked as ModelProps. It would be a breaking change, but I don't think it would affect  many people. Any ideas?